managed/services/agents/vmagent.go

```diff
@@ -31,48 +32,103 @@ var (
 	maxScrapeSizeDefault = "64MiB"
 )
 
+// extractCredentialsFromURL extracts username and password from a URL string.
```

Comment:
Are comments for not exposed func needed? I mena do we want to keep them? Since func naming is good and clear. Same applies for rest.